frontend/src/modules/WellCompletion/wellCompletionsDataAccessor.ts

```diff
@@ -0,0 +1,236 @@
+        this._searchWellText = searchWell;
+    }
+
+    setHideZeroCompletions(hideZeroCompletions: boolean): void {
+        this._hideZeroCompletions = hideZeroCompletions;
+    }
+
+    getTimeSteps(): string[] {
+        if (this._dataSet === undefined) return [];
+        return this._dataSet.timeSteps;
+    }
+
+    createPlotData(timeStepSelection: string | [string, string], timeAggregation: TimeAggregation): PlotData | null {
+        // TODO: Consider removing function arguments, and use setter-methods for each argument and set to an attribute.
+        //       This would make it easier to modify/adjust single attributes and call "createPlotData" again.
+        if (!this._dataSet) return null;
+
+        let timeStepIndexRange: [number, number] | null = null;
+        if (typeof timeStepSelection === "string") {
+            const timeStepIndex = this._dataSet ? this._dataSet.timeSteps.indexOf(timeStepSelection) : 0;
+            timeStepIndexRange = [timeStepIndex, timeStepIndex];
+        } else {
+            timeStepIndexRange = [
+                this._dataSet.timeSteps.indexOf(timeStepSelection[0]),
+                this._dataSet.timeSteps.indexOf(timeStepSelection[1]),
+            ];
+        }
+        if (!timeStepIndexRange) return null;
+
+        // Filter wells based on search text
+        // TODO: Add filtering of attributes as well
+        const wellNameRegex = getRegexPredicate(this._searchWellText);
+        const filteredWells = this._searchWellText
+            ? Array.from(this._wells as WellCompletionWell_api[]).filter((well) => wellNameRegex(well.name))
+            : this._wells;
+
+        return this.computePlotData(
+            this._subzones,
+            filteredWells,
+            timeStepIndexRange,
+            timeAggregation,
+            this._hideZeroCompletions,
+            this._dataSet?.units
+        );
+    }
+
+    private findEarliestCompletionDateIndices(
+        well: WellCompletionWell_api,
+        subzones: WellCompletionZone_api[]
+    ): number {
+        //The earliest completion date for the given well
+        let earliestCompDateIndex = Number.POSITIVE_INFINITY;
+        subzones.forEach((zone) => {
+            if (zone.name in well.completions) {
+                const completion = well.completions[zone.name];
+                //Find the earliest date for the given completion
+                const earliestDate = completion.t.find((_, index) => completion.open[index] > 0);
+                if (earliestDate !== undefined) earliestCompDateIndex = Math.min(earliestCompDateIndex, earliestDate);
+            }
+        });
+
+        return earliestCompDateIndex;
+    }
+
+    /**
+     * Util method to prepare stratigraphy and well data from the given time step range and other settings for plotting
+     * @param subzones
+     * @param wells
+     * @param range
+     * @param timeAggregation
+     * @param hideZeroCompletions
+     * @returns
```

Comment:
Sometimes I see this doc style and sometimes there is none or another Should we keep only the description?

Reply:
Removed